clients/src/main/java/org/apache/kafka/common/protocol/ProtoUtils.java

```diff
@@ -27,7 +27,7 @@ private static Schema schemaFor(Schema[][] schemas, int apiKey, int version) {
         if (apiKey < 0 || apiKey > schemas.length)
             throw new IllegalArgumentException("Invalid api key: " + apiKey);
         Schema[] versions = schemas[apiKey];
-        if (version < 0 || version > versions.length)
+        if (version < 0 || version >= versions.length)
             throw new IllegalArgumentException("Invalid version for API key " + apiKey + ": " + version);
         if (versions[version] == null)
```

Comment:
Could the check above be removed all together? This check looks like it would cover all cases. Even if there is a gap or jump in versions.